### PR TITLE
create model

### PR DESCRIPTION
**Pull request by TRONGTAI1801**

### CODE REVIEW

File: app/src/main/java/com/framgia/music/model/Currently.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import android.os.Parcelable;
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Currently implements Parcelable{
+    private Weather mWeather;
+
+    public Currently(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Currently.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Daily.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.ArrayList;
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Daily implements Parcelable{
+    private ArrayList<Weather> mWeathers;
+
+    public Daily(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Daily.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Hourly.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Hourly implements Parcelable{
+
+    private ArrayList<Weather> mWeather;
+
+    public Hourly(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Hourly.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Humidity.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Humidity implements Parcelable {
+
+    private String mHumidity;
+
+    public Humidity(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Humidity.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Location.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+public class Location implements Parcelable {
+    private String mLongitude;
+    private String mLatitude;
+    private String mTimezone;
+    private Currently mCurrently;
+    private Hourly mHourly;
+    private Daily mDaily;
+
+    public Location(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Location.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Location.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        mLongitude = longitude;
+        mLatitude = latitude;
+        mTimezone = timezone;
+        mCurrently = currently;
+        mHourly = hourly;
+        mDaily = daily;
+    }
+
+
+    public Location(Location location){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Location.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Temperature.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Temperature implements Parcelable {
+
+    private String mTemperature;
+    private String mTemperatureHight;
+    private String mTemperatureLow;
+
+    public Temperature(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Temperature.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/UVIndex.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class UVIndex implements Parcelable {
+
+    private String mUVIndex;
+
+    public UVIndex(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/UVIndex.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/UVIndex.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        mUVIndex = "";
+    }
+
+    public UVIndex(String UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/UVIndex.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Visibility.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Visibility implements Parcelable {
+
+    private String mVisibility;
+
+    public Visibility(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Visibility.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+public class Weather implements Parcelable {
+
+    private String mIcon;
+    private Temperature mTemperature;
+    private Wind mWind;
+    private Humidity mHumidity;
+    private Visibility mVisibility;
+    private UVIndex mUVIndex;
+
+    public Weather(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        mTemperature = new Temperature();
+        mWind = new Wind();
+        mHumidity = new Humidity();
+        mVisibility = new Visibility();
+        mUVIndex = new UVIndex();
+        mIcon = "";
+    }
+
+    public Weather(String icon, Temperature temperature, Wind wind, Humidity humidity, Visibility visibility,
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Line is longer than 100 characters (found 109).

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+            UVIndex UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+
+    public void setVisibility(Visibility visibility) {
+        mVisibility = visibility;
+    }
+
+    public UVIndex getUVIndex() {
+        return mUVIndex;
+    }
+
+    public void setUVIndex(UVIndex UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Wind.java [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Wind implements Parcelable{
+
+    private String mSpeed;
+
+    public Wind(){
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Wind.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/model/Wind.java [hunk 16]
```diff
@@ -0,0 +1,51 @@
+package com.framgia.music.model;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Wind implements Parcelable{
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Wind.java]: WARNING: [ShortClassName] Avoid short class names like Wind
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/framgia/music/model/UVIndex.java [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class UVIndex implements Parcelable {
+
+    private String mUVIndex;
+
+    public UVIndex(String UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/UVIndex.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+public class Weather implements Parcelable {
+
+    private String mIcon;
+    private Temperature mTemperature;
+    private Wind mWind;
+    private Humidity mHumidity;
+    private Visibility mVisibility;
+    private UVIndex mUVIndex;
+
+    public Weather(String icon, Temperature temperature, Wind wind, Humidity humidity, Visibility visibility,
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Line is longer than 100 characters (found 109).

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+            UVIndex UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/UvIndex.java [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class UvIndex implements Parcelable {
+
+    private String mUvIndex;
+
+    public UvIndex(String UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/UvIndex.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+
+    private String mIcon;
+    private Temperature mTemperature;
+    private WindSpeed mWind;
+    private Humidity mHumidity;
+    private Visibility mVisibility;
+    private UvIndex mUvIndex;
+
+    public Weather(String icon, Temperature temperature, WindSpeed wind,
+            Humidity humidity, Visibility visibility, UvIndex UVIndex) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: Name 'UVIndex' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/music/model/Daily.java [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import java.util.ArrayList;
+
+/**
+ * Created by trong_tai on 25/01/2018.
+ */
+
+public class Daily implements Parcelable {
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/music/model/Daily.java]: Daily và Hourly giống nhau, sao e không gộp lại thành 1

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    @Override
+    public void writeToParcel(Parcel parcel, int i) {
+        parcel.writeString(mIcon);
+        parcel.writeParcelable(mTemperature, i);
+        parcel.writeParcelable(mHumidity, i);
+        parcel.writeParcelable(mVisibility, i);
+        parcel.writeParcelable(mUvIndex, i);
+        parcel.writeParcelable(mTimeWeather,i);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/music/model/Weather.java [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+        parcel.writeParcelable(mWindSpeed,i);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/Weather.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/com/framgia/music/model/TimeWeather.java [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+/**
+ * Created by trong_tai on 30/01/2018.
+ */
+
+public class TimeWeather implements Parcelable{
+    private String timeWeather;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/model/TimeWeather.java]: WARNING: [AvoidFieldNameMatchingTypeName] It is somewhat confusing to have a field name matching the declaring class name
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingTypeName

File: gradlew [hunk 26]
```diff
@@ -74,14 +74,14 @@ if [ -n "$JAVA_HOME" ] ; then
         die "ERROR: JAVA_HOME is set to an invalid directory: $JAVA_HOME
 
 Please set the JAVA_HOME variable in your environment to match the
-location of your Java installation."
+locationWeather of your Java installation."
```
Comment (daolq3012) [inline gradlew]: Pls keep old code!

File: gradlew [hunk 27]
```diff
@@ -74,14 +74,14 @@ if [ -n "$JAVA_HOME" ] ; then
...
     fi
 else
     JAVACMD="java"
     which java >/dev/null 2>&1 || die "ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
 
 Please set the JAVA_HOME variable in your environment to match the
-location of your Java installation."
+locationWeather of your Java installation."
```
Comment (daolq3012) [inline gradlew]: Pls keep old code!

File: gradlew.bat [hunk 28]
```diff
@@ -27,7 +27,7 @@ echo.
 echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
 echo.
 echo Please set the JAVA_HOME variable in your environment to match the
-echo location of your Java installation.
+echo locationWeather of your Java installation.
```
Comment (daolq3012) [inline gradlew.bat]: Pls keep old code!

File: gradlew.bat [hunk 29]
```diff
@@ -41,7 +41,7 @@ echo.
 echo ERROR: JAVA_HOME is set to an invalid directory: %JAVA_HOME%
 echo.
 echo Please set the JAVA_HOME variable in your environment to match the
-echo location of your Java installation.
+echo locationWeather of your Java installation.
```
Comment (daolq3012) [inline gradlew.bat]: Pls keep old code!

**Pull request merged**